zio/vngio/reader.go

```diff
@@ -6,12 +6,29 @@ import (
 	"os"
 
 	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/compiler/optimizer/demand"
 	"github.com/brimdata/zed/vector"
 	"github.com/brimdata/zed/vng"
 	"github.com/brimdata/zed/zio"
 )
 
+type ReaderOpts struct {
+	Demand demand.Demand
+}
```

Comment:
In my dream world, all of the zio.Reader constructors will accept a demand.Demand, so rather than put it in here, I'm in favor of making it a NewReader parameter, like so:
```
func NewReader(zctx *zed.Context, r io.Reader, d demand.Demand) (zio.Reader, error)
```

Reply:
Should it be in anyio.ReaderOpts then? Or as an extra argument to anyio.NewReader too? Eventually it's going to get passed down from the dag in kernel.compileLeaf.

Reply:
I like it as an extra argument to anyio.NewReader.

Reply:
https://github.com/brimdata/zed/pull/4850/commits/2c177b19eccd0c81a6994cd0b4678cb7c0f15ac1